deployment/src/test/java/TestTriggers.java

```diff
@@ -0,0 +1,88 @@
+import com.impossibl.postgres.jdbc.PGDataSource;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.junit.Assume;
+import org.junit.jupiter.api.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLException;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class DatabaseTests {
+  private static final File initSqlScriptFile = new File("postgres-init-db/sql/merlin/init.sql");
+  private java.sql.Connection connection;
+
+  @BeforeAll
+  void beforeAll() throws SQLException, IOException, InterruptedException {
+
+    // Create test database and grand privileges
+    ProcessBuilder pb = new ProcessBuilder("psql",
+                                           "postgresql://postgres:postgres@localhost:5432",
+                                           "-v", "ON_ERROR_STOP=1",
+                                           "-c", "CREATE DATABASE aerie_merlin_test;",
+                                           "-c", "GRANT ALL PRIVILEGES ON DATABASE aerie_merlin_test TO aerie;"
+    );
+
+    Process proc = pb.start();
+
+    // Handle the case where we cannot connect to postgres by skipping the tests
+    final String errors = new String(proc.getErrorStream().readAllBytes(), StandardCharsets.UTF_8);
+    Assume.assumeFalse(errors.contains("could not connect to server: Connection refused"));
+    proc.waitFor();
+    proc.destroy();
+
+    // Grant table privileges to aerie user for the tests
+    // Apparently, the previous privileges are insufficient on their own
+    pb = new ProcessBuilder("psql",
```

Comment:
This seems like a good spot for a pattern I've seen around the aerie codebase - in order to avoid re-assigning the `pb` variable to mean two different things at different points in this methods, we can wrap lines 21-34 in a "block" (curly braces) to limit the scope of variable declarations in that block.

This way, we can mark each local variable as `final`, yet still reuse the same variable name.

Also: the aerie style preference is to use `var` wherever possible, instead of writing the type twice.

Let me know if that didn't make sense :+1:

I've sketched out what this might look like here: https://github.com/mattdailis/aerie/commit/52a29b9105b6755639dfc3a9d8e431de1dc63eee

Reply:
I was using shadowing intentionally here, I can adjust that to avoid doing so.

Reply:
What is the performance hit in Java for having a variable sized type by using var?

Reply:
Ah, just read that it just signals type inference, not variable sized types.

Reply:
I think Shadowing and blocks are both trying to achieve the same goal, which is to make sure only one instance of that type is available at certain parts of the method.

I think blocks are nice because they make the reader's job easier, since we can safely assume that the variable is out of scope after the closing brace. Shadowing does that too, but it takes a bit more focus on the part of the reader to notice that it's being reassigned (especially if there's more than on variable involved).

The most important "style preference" of the ones I mentioned is to use `final` everywhere we possibly can, since that is what really helps the reader relax and not worry about reassignment. In this case, a block would just allow us to use the same variable name twice, and mark both as `final`.

Shadowing is somewhat incompatible with final... I could be swayed that in some cases it's better to shadow than to use `final`, but our default is to make each variable and parameter `final` unless there's a good reason not to (there certainly are cases when we reassign variables, especially in loops).

Like you pointed out, `var` is just shorthand for "use the type of the right hand side of this assignment". It's just to reduce verbosity (and in some cases, it can reduce the number of places that code needs to be changed if we change the name of a class). I feel less strongly about this one, but I've had @pcrosemurgy request this change on my PRs.